### PR TITLE
Add event type filter to ctapipe-process

### PR DESCRIPTION
**Pull request by maxnoe**
Allows filtering out un-wanted event types form the output.

### CODE REVIEW

**Review by kosack — DISMISSED** [review]

File: ctapipe/utils/event_type_filter.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class EventTypeFilter(Component):
+    """Check that an event has one of the allowed types"""
+
+    allowed_types = Set(
+        # add both the enum instance and the integer values to support
+        # giving the integers in config files.
+        trait=Enum(list(EventType) + [t.value for t in EventType]),
+        default_value=None,
+        allow_none=True,
+        help="The allowed types. Set to None to allowe all types.",
```
Comment (LukasNickel) [inline ctapipe/utils/event_type_filter.py:18]:
```suggestion
        help="The allowed types. Set to None to allow all types.",
```

**Review by LukasNickel — DISMISSED** [review]

File: ctapipe/tools/process.py [hunk 2]
```diff
@@ -238,6 +245,7 @@ def start(self):
         self.log.info("(re)compute DL2: %s", self.should_compute_dl2)
         self.event_source.subarray.info(printer=self.log.info)
 
+        event = None
```
Comment (nbiederbeck) [inline ctapipe/tools/process.py:248]: Why this line? Everything else inside this function is inside the for-loop, where `event` gets overwritten

Comment (maxnoe) [reply ctapipe/tools/process.py:248]: It is for the case that the source has no triggered events that `event` is not undefined.

Comment (maxnoe) [reply ctapipe/tools/process.py:248]: Ah, sorry, it's actually not needed in this PR, comes from other code I was also adding

Comment (nbiederbeck) [reply ctapipe/tools/process.py:248]: Ok then, can stay here I think

**Review by nbiederbeck — DISMISSED** [review]

**Review by nbiederbeck — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**